pennylane/transforms/core/transform_dispatcher.py

```diff
@@ -623,27 +496,170 @@ def final_transform(self) -> bool:
         return self._transform_dispatcher.final_transform
 
 
-def _create_transform_primitive(name):
-    try:
-        # pylint: disable=import-outside-toplevel
-        from pennylane.capture.custom_primitives import QmlPrimitive
-    except ImportError:
-        return None
+@TransformDispatcher.generic_register
+def _apply_to_tape(obj: QuantumScript, transform, *targs, **tkwargs):
```

Comment:
With `singledispatchmethod`, all these registrations can happen inside `TransformDispatcher`, which I since it's more consistent with the OOP paradigm.